ksef-client/src/main/java/io/alapierre/ksef/batch/BatchHelper.java

```diff
@@ -0,0 +1,261 @@
+import org.apache.commons.io.input.BoundedInputStream;
+import pl.akmf.ksef.sdk.api.builders.batch.OpenBatchSessionRequestBuilder;
+import pl.akmf.ksef.sdk.client.interfaces.CryptographyService;
+import pl.akmf.ksef.sdk.client.interfaces.KSeFClient;
+import pl.akmf.ksef.sdk.client.model.ApiException;
+import pl.akmf.ksef.sdk.client.model.session.*;
+import pl.akmf.ksef.sdk.client.model.session.batch.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.DigestOutputStream;
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * @author Adrian Lapierre {@literal al@alapierre.io}
+ * Copyrights by original author 26.11.2025
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class BatchHelper {
+
+    private final CryptographyService cryptographyService;
+    private final KSeFClient client;
+
+    /**
+     * Przygotowuje paczkę wsadową: tworzy archiwum ZIP z faktur, dzieli je na części i szyfruje.
+     *
+     * @param source źródło faktur (np. katalog, lista w pamięci)
+     * @param config konfiguracja (katalog wyjściowy, rozmiar części, czyszczenie)
+     * @return wynik przygotowania zawierający ścieżki do zaszyfrowanych części, klucze i hashe
+     * @throws BatchProcessingException w przypadku błędów IO lub kryptograficznych
+     */
+    public BatchResult prepareBatch(InvoiceSource source, BatchConfig config) {
+
+        // 1. Utwórz ZIP i zbierz hashe faktur (zapis do pliku tymczasowego)
+        ZipContext zipContext = createZipWithHashes(source);
+        File plainZipFile = zipContext.file();
+        log.info("Utworzono plik ZIP: {}, rozmiar: {}", plainZipFile.getAbsolutePath(), plainZipFile.length());
+
+        List<BatchPartInfo> encryptedParts = new ArrayList<>();
+        byte[] aesKey;
+        final String encryptedCipherKey;
+        byte[] iv;
+        String zipHash;
+        long zipSize = 0;
+
+        try {
+            // 2. Pobierz dane kryptograficzne (klucz AES i IV)
+            EncryptionData encryptionData = cryptographyService.getEncryptionData();
+            aesKey = encryptionData.cipherKey();
+            iv = encryptionData.cipherIv();
+            encryptedCipherKey = encryptionData.encryptedCipherKey();
+
+            // 3. Oblicz hash całego pliku ZIP
+            try (InputStream is = new FileInputStream(plainZipFile)) {
+                FileMetadata meta = cryptographyService.getMetaData(is);
+                zipHash = meta.getHashSHA();
+                zipSize = meta.getFileSize();
+            }
+            log.info("ZIP Meta: size={}, hash={}", zipSize, zipHash);
+
+            // 4. Podziel i zaszyfruj ZIP kawałkami (byte[])
+            try (InputStream fis = new FileInputStream(plainZipFile)) {
+
+                byte[] buffer = new byte[(int) config.maxPartSize()];
```

Comment:
`maxPartSize` jest typu long, a tutaj pastujemy to do inta. Może spowodować to problem.